src/PrizeVault.sol

```diff
@@ -781,18 +789,36 @@ contract PrizeVault is TwabERC20, Claimable, IERC4626, ILiquidationSource, Ownab
     /// @dev The yield fee balance is included since it's cheaper to keep track of those shares
     ///      internally instead of doing an additional TWAB mint on every liquidation.
     /// @param _totalSupply The total share supply of the vault
+    /// @param _yieldFeeBalance The unrealized yield fee balance
     /// @return The total asset debt of the vault
-    function _totalDebt(uint256 _totalSupply) internal view returns (uint256) {
-        return _totalSupply + yieldFeeBalance;
+    function _totalDebt(uint256 _totalSupply, uint256 _yieldFeeBalance) internal pure returns (uint256) {
+        return _totalSupply + _yieldFeeBalance;
     }
 
     /// @notice Returns the remaining supply that can be minted without exceeding the TwabController limits.
     /// @dev The TwabController limits the total supply for each vault to uint96
+    /// @dev The yield fee balance is included to account for the unrealized shares
     /// @param _totalSupply The total share supply of the vault
+    /// @param _yieldFeeBalance The unrealized yield fee balance
     /// @return The remaining supply that can be minted without exceeding TWAB limits
-    function _twabSupplyLimit(uint256 _totalSupply) internal pure returns (uint256) {
+    function _twabSupplyLimit(uint256 _totalSupply, uint256 _yieldFeeBalance) internal pure returns (uint256) {
         unchecked {
-            return type(uint96).max - _totalSupply;
+            return type(uint96).max - (_totalSupply + _yieldFeeBalance);
+        }
+    }
+
+    /// @notice Verifies that the resulting TWAB supply limit can support the minting of the full yield fee balance.
+    /// @dev Reverts if the TWAB supply limit is exceeded.
+    /// @dev This MUST be called anytime there is a positive increase in the net total of minted shares and yield
+    /// fee balance.
+    /// @param _totalSupply The total share supply of the vault
+    /// @param _yieldFeeBalance The unrealized yield fee balance
+    function _enforceTwabSupplyLimit(uint256 _totalSupply, uint256 _yieldFeeBalance) internal pure {
```

Comment:
It seems like this function introduces a new concept of "unrealized" here... it took me a second to understand what was going on.  I feel like it would be more clearly articulated by leaning into the idea of the "mint limit", and just ensuring new shares don't exceed the mint limit.

For regular minting, this check is really just a human-friendly error code (instead of actual overflow).  For the yield fee, however, it *really* needs to be caught ahead of time before the mint

So i feel like the "newShares" could be a parameter and it just checks that `newShares < _mintLimit`